### ISSUE TITLE
ocsstore: kio error: missing handler popup

### ISSUE THREAD

**Issue by star-buck** [blue-systems/netrunner-issues-old#153]
is there anything that can be done on netrunner side to prevent this message from popping up:
![screenshot_20181231_094825](https://user-images.githubusercontent.com/840070/50557676-bf6bc300-0ce7-11e9-9b0c-f6746fa4c19e.png)

this didnt come up on 18.03 netrunner, and is caused by using latest opendesktop appimage, which is still working fine despite this message.
Maybe there is some config where kio defines these and one could alter it to prevent the message?

**Comment by star-buck** [comment]
appimage can be downloaded here for trying it:
https://www.opendesktop.org/p/1175480/

**Comment by NuLogicSystems** [comment]
I just downloaded the appimage and chose "run once" on my Netrunner Rolling install, I did not get that error message.
Appimage: ocs-store-3.5.0-1-x86_64.AppImage
Appimagelauncher from manjaro: 1.0alpha+git20181009.b04daee-1
libappimage from arch: 0.1.2-2

**Comment by star-buck** [comment]
Where did you run it on?

**Comment by NuLogicSystems** [comment]
On my Netrunner Rolling install.
However, Manjaro hasn't updated Appimagelauncher since 20181009.

**Comment by NuLogicSystems** [comment]
It also looks like that appimage was just updated today (2018-12-3) on opendesktop.org, just  2 hours ago: version 3.5.0
https://www.opendesktop.org/p/1175480/#updates-panel

**Comment by star-buck** [comment]
Yes, its still not working on new debian 1901, i tested this morning.
The question is why is it working on Rolling, as in what might be missing
package or different config between the two that it works on your end and
not here?

@James: Can you try without appimagelauncher, like completey remove AIL and
execute once by doubleclick on it?

Maybe appimagelauncher does something and since i dont have AIL on debian
working, i cant compare.

**Comment by chinmoyrp** [comment]
I was getting the same error on my KDE neon VM and installing ocs-url (https://store.kde.org/p/1136805) fixed it. @star-buck  Can you give it a try?

**Comment by NuLogicSystems** [comment]
Removed AIL, same result, ocs-store-3.5.0-1-x86_64.AppImage launches just fine.

@chinmoyrp,
I didn't even think about that.
I have ocs-url 3.0.3 from our rolling repository installed.

**Comment by NuLogicSystems** [comment]
@chinmoyrp,
Nope, after removing ocs-url, there still no error executing ocs-store-3.5.0-1-x86_64.AppImage.

**Comment by chinmoyrp** [comment]
Removing /usr/share/applications/ocs-url.desktop definitely shows an error here. Maybe there's some other handler calling the ocs-url binary?

**Comment by star-buck** [comment]
Can one simulate ocsurl when we dont have a package yet?

Still the appimage of ocsstore should already have ocsurl as well included,
hence installing it additional would be redundant.

**Comment by chinmoyrp** [comment]
@NuLogicSystems Internally KIO searches for any desktop file with MimeType=x-scheme-handler/ocs entry and it lands a hit when ocs-url is installed. So most probably there's some other service (or something  cached) in your installation with the said entry.

**Comment by NuLogicSystems** [comment]
@chinmoyrp,
Never mind, I do sincerely apologize.
I now see where my confusion came from, I didn't understand when the error occures.
Trying to install anything inside ocs-store is now giving me this error.
Re-installing ocs-url does indeed fix the problem, but then I end up with both dialogs:
![screenshot_20181231_085217](https://user-images.githubusercontent.com/10186587/50561509-7745a400-0cd9-11e9-9ac2-9065d180f037.png)

**Comment by NuLogicSystems** [comment]
Since the appimage is a self contained environment, shouldn't this mimetype be set inside of it?

**Comment by chinmoyrp** [comment]
> Can one simulate ocsurl when we dont have a package yet? 

Simulating the download part is easy but install gets little tricky.

>Still the appimage of ocsstore should already have ocsurl as well included, hence installing it additional would be redundant.

I don't think it is there. I extracted the appimage and couldn't locate the ocs-url binary.

>Since the appimage is a self contained environment, shouldn't this mimetype be set inside of it?

It should be and that would mean including ocs-url as well in the appimage.

**Comment by chinmoyrp** [comment]
> @chinmoyrp,
> Never mind, I do sincerely apologize.
> I now see where my confusion came from, I didn't understand when the error occures.
> Trying to install anything inside ocs-store is now giving me this error.
> Re-installing ocs-url does indeed fix the problem, but then I end up with both dialogs:
> ![screenshot_20181231_085217](https://user-images.githubusercontent.com/10186587/50561509-7745a400-0cd9-11e9-9ac2-9065d180f037.png)

The frontend shouldn't really parse any url here since the store website invokes xdg-open. I think this should be treated as a bug in ocs-store(?)

**Comment by NuLogicSystems** [comment]
@chinmoyrp,

> The frontend shouldn't really parse any url here since the store website invokes xdg-open. I think this should be treated as a bug in ocs-store(?)

I agree, an appimage shouldn't need an external application, or mimetype to function properly.
However, this brings up another issue. If the ocs-store appimage doesn't create a desktop file, then using a browser like Firefox would not be able to set a mime type (Launch Application) for the link (url) either.
![screenshot_20181231_102441](https://user-images.githubusercontent.com/10186587/50562909-60597e80-0ce6-11e9-994e-eb71ded2aff0.png)

> It should be and that would mean including ocs-url as well in the appimage.

Not really since ocs-url, and ocs-store, both have the same functionality. Hence the two "do you want to install" dialog's in my previous posts screenshot. One natively installed ocs-url, and the other part ocs-store itself (inside the appimage window).

**Comment by star-buck** [comment]
@chinmoyrp : any idea how to not make it popup? as it does work fine after closing the error message and simply continue with the installation, e.g. try installing a wallpaper and find it properly installed and ready to be set from the app.

**Comment by chinmoyrp** [comment]
@star-buck The popup won't appear if we somehow prevent xdg-open(thereby prevent kioclient) from opening the url. I believe it has to be done from within ocs-store though I am not knowledgeable in electron and js so I can't say for sure if it's possible.
Here : https://git.opendesktop.org/akiraohgaki/ocs-store/blob/master/app/scripts/renderers/ipc-renderer.js#L74 is the event handling code for ocs:/ url. This is where the prevention should happen.

Also installing ocs-url does fix the issue but it creates the problem of redundant popups. The easiest workaround is to add a dummy schema handler that will eat the popup.

**Comment by star-buck** [comment]
Add a dummy schema handler which will eat the popup inside ocsstore
appimage?
Or on Netrunner 1901?

Any example schema/file that Akira or Netrunner can use?

**Comment by chinmoyrp** [comment]
In Netrunner 1901.
Something like this:

_/usr/share/applications/ocs-dummy.desktop_
>[Desktop Entry]
Name=ocs-url
Exec=true
Type=Application
Terminal=false
NoDisplay=true
MimeType=x-scheme-handler/ocs;x-scheme-handler/ocss;

**Comment by star-buck** [comment]
Thanks Chinmoy, will try out in 1901.

Since it doesnt seem to fail in Ubuntu 18.04 Gnome, the question is if this
would have to be done in any KDE Plasma distro with kio and no ocsurl, or
if Akira can do something like this within the Appimage to solve it
pre-emptively?

**Comment by chinmoyrp** [comment]
>if this would have to be done in any KDE Plasma distro with kio and no ocsurl

IMO unless users have ocs-store installed this shouldn't be used.

>or if Akira can do something like this within the Appimage to solve it pre-emptively?

I have been trying to make this 'hack' work from within the appimage but so far I have had no success. So I can't say for sure.
